lib/unix/git_unix.ml

```diff
@@ -244,9 +243,14 @@ module IO_FS = struct
   let read_file file =
     Lwt_pool.use openfile_pool (fun () ->
         Log.info "Reading %s" file;
-        Lwt_io.(with_file ~mode:input) ~flags:[Unix.O_RDONLY] file (fun ch ->
+        Lwt_unix.stat file >>= fun stats ->
```

Comment:
maybe the LargeFile size? http://ocsigen.org/lwt/2.5.1/api/Lwt_unix.LargeFile otherwise 2GB limit..